tests/unit/resource/suspend_pool.cpp

```diff
@@ -0,0 +1,160 @@
+//  Copyright (c) 2017 Mikael Simberg
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// Simple test verifying basic resource_partitioner functionality.
+
+#include <hpx/hpx_start.hpp>
+#include <hpx/include/async.hpp>
+#include <hpx/include/resource_partitioner.hpp>
+#include <hpx/include/threadmanager.hpp>
+#include <hpx/include/threads.hpp>
+#include <hpx/runtime/threads/executors/pool_executor.hpp>
+#include <hpx/runtime/threads/policies/scheduler_mode.hpp>
+#include <hpx/runtime/threads/policies/schedulers.hpp>
+#include <hpx/runtime/threads/thread_helpers.hpp>
+#include <hpx/util/lightweight_test.hpp>
+
+#include <atomic>
+#include <cstddef>
+#include <string>
+#include <utility>
+#include <vector>
+
+int hpx_main(int argc, char* argv[])
+{
+    try
+    {
+        hpx::this_thread::get_pool()->suspend();
+        HPX_TEST_MSG(false, "Suspending is not possible on own pool");
+    }
+    catch (std::runtime_error const&)
+    {
+    }
```

Comment:
Where is the `std::runtime_error` thrown? If this is thrown by HPX it should be changed to using an `hpx::exception` instead.

Either way, it looks like the exception is expected in this test case, if so, please add a bool checking that it was indeed thrown:

    bool exception_thrown = false;
    try { ... } catch(hpx::exception const& e) { exception_thrown = true; }
    HPX_TEST(exception_thrown);

Reply:
Hmm, yes, it is thrown by HPX. I did not realize it will be an `hpx::exception` then... Will update it the way you've suggested.

Reply:
Thanks, LGTM.